src/hpc/aobjects.c

```diff
@@ -994,7 +993,7 @@ Obj GetTLRecordField(Obj record, UInt rnam)
   contents = GetTLInner(record);
   table = ADDR_OBJ(contents);
   tlrecord = table[TLR_DATA+TLS(threadID)];
-  if (!tlrecord || !FindPRec(tlrecord, rnam, &pos, 1)) {
+  if (!tlrecord || !(pos = PositionPRec(tlrecord, rnam, 1))) {
```

Comment:
I dislike assignments in conditions. Could this perhaps be rewritten to something like this?
```C
  pos = tlrecord ? PositionPRec(tlrecord, rnam, 1) : 0;
  if (pos) {
```

Reply:
I agree. I don't like it either -- but I also didn't like turning it into a double level if. I'll try something like that.